src/ast/semantic_analyser.cpp

```diff
@@ -559,6 +559,17 @@ void SemanticAnalyser::visit(Call &call)
   else if (call.func == "ustack") {
     check_stack_call(call, Type::ustack);
   }
+  else if (call.func == "strncmp") {
+    check_nargs(call, 3);
```

Comment:
To prevent going out of range: 

```
    if(check_nargs(call, 3)) {
      check_arg(call, Type::string, 0);
      check_arg(call, Type::string, 1);
      check_arg(call, Type::integer, 2, true);

     Integer &size = static_cast<Integer&>(*call.vargs->at(2));
     if (size.n < 0)
       err_ << "Builtin strncmp requires a non-negative size" << std::endl; 
   }
    call.type = SizedType(Type::integer, 8);
  }
```

Reply:
```
bpftrace -e 'i:s:1 { strncmp("a"); }'
terminate called after throwing an instance of 'std::out_of_range'
  what():  vector::_M_range_check: __n (which is 1) >= this->size() (which is 1)
Aborted
```